src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+const body = document.querySelector('body');
+const div = document.createElement('div');
+let leftClickHappened = false;
+let rightClickHappened = false;
+
+div.setAttribute('data-qa', 'notification');
+body.appendChild(div);
+
+const showNotification = (message, type) => {
+  const notification = document.createElement('div');
+
+  notification.className = type;
+  notification.setAttribute('data-qa', 'notification');
+  notification.textContent = message;
+  body.appendChild(notification);
+};
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleLeftClick = (e) => {
+    if (e.button === 0) {
+      resolve(`First promise was resolved`);
+      document.removeEventListener('mouseup', handleLeftClick);
+    }
+  };
+
+  document.addEventListener('mouseup', handleLeftClick);
+
+  setTimeout(() => {
+    reject(new Error(`First promise was rejected`));
+    document.removeEventListener('mouseup', handleLeftClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve(`Second promise was resolved`);
+      document.removeEventListener('mouseup', handleClick);
+    }
```

Comment:
Remember to prevent the default context menu from appearing when the right mouse button is clicked, since it can interfere with the intended behavior of the application. You can use 'e.preventDefault()' in the event handler when 'e.button === 2'.